tests/dt.go

```diff
@@ -0,0 +1,250 @@
+	DownPeers       []*pdpb.PeerStats `json:"down_peers,omitempty"`
+	PendingPeers    []*metapb.Peer    `json:"pending_peers,omitempty"`
+	WrittenBytes    uint64            `json:"written_bytes,omitempty"`
+	ReadBytes       uint64            `json:"read_bytes,omitempty"`
+	ApproximateSize int64             `json:"approximate_size,omitempty"`
+	ApproximateKeys int64             `json:"approximate_keys,omitempty"`
+}
+
+// RegionsInfo contains some regions with the detailed region info.
+type RegionsInfo struct {
+	Count   int           `json:"count"`
+	Regions []*RegionInfo `json:"regions"`
+}
+
+func (oa *operatorActions) LabelNodes() error {
+	nodes, err := oa.kubeCli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for i, node := range nodes.Items {
+		index := i % RackNum
+		node.Labels[RackLabel] = fmt.Sprintf("rack%d", index)
+		_, err = oa.kubeCli.CoreV1().Nodes().Update(&node)
+		if err != nil {
+			glog.Errorf("label node:[%s] failed!", node.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func (oa *operatorActions) LabelNodesOrDie() {
+	err := oa.LabelNodes()
+	if err != nil {
+		slack.NotifyAndPanic(err)
+	}
+}
+
+func (oa *operatorActions) CheckDisasterTolerance(cluster *TidbClusterConfig) error {
+	nodeMap := map[string]corev1.Node{}
+	nodes, err := oa.kubeCli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes.Items {
+		nodeMap[node.Name] = node
+	}
+
+	pds, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).PD().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	err = oa.checkPodsDisasterTolerance(pds.Items, nodeMap)
+	if err != nil {
+		return err
+	}
+
+	tikvs, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).TiKV().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	err = oa.checkPodsDisasterTolerance(tikvs.Items, nodeMap)
+	if err != nil {
+		return err
+	}
+
+	tidbs, err := oa.kubeCli.CoreV1().Pods(cluster.Namespace).List(
+		metav1.ListOptions{LabelSelector: labels.SelectorFromSet(
+			label.New().Instance(cluster.ClusterName).TiDB().Labels(),
+		).String()})
+	if err != nil {
+		return err
+	}
+	return oa.checkPodsDisasterTolerance(tidbs.Items, nodeMap)
+}
+
+func (oa *operatorActions) checkPodsDisasterTolerance(allPods []corev1.Pod, nodeMap map[string]corev1.Node) error {
+	rackPods := map[string][]corev1.Pod{}
+	for _, pod := range allPods {
+		if node, exist := nodeMap[pod.Spec.NodeName]; exist {
+			pods, exist := rackPods[node.Labels[RackLabel]]
+			if !exist {
+				pods = []corev1.Pod{}
+			}
+			pods = append(pods, pod)
+			rackPods[node.Labels[RackLabel]] = pods
+		}
+	}
+
+	podNum := len(allPods)
+	maxPodsOneRack := podNum / RackNum
```

Comment:
If podNum is 5 RackNum is 3, mod will be 2 and maxPodsOneRack will be 2... if this test will be pass, the podNumOnRack should more than 2 and less than 2. It must be failed.maybe the 
podNumOnRack < mod should change to 
podNumOnRack < podNum / RackNum ?

Reply:
yes, I make a mistake